File: https:/github.com/Raushankumar0720/Leetcode/7-ReverseInteger/7-ReverseInteger.cpp
// Last updated: 7/6/2026, 11:44:13 PM
class Solution {
public:
    int reverse(int x) {
        long long rev = 0;  // Use long long to detect overflow during calculation

        while (x != 0) {
            int digit = x % 10;
            x /= 10;

            rev = rev * 10 + digit;

            // Check overflow: must stay within 32-bit signed integer range
            if (rev > INT_MAX || rev < INT_MIN) {
                return 0;
            }
        }

        return (int)rev;
    }
};

File: https:/github.com/Raushankumar0720/Leetcode/7-ReverseInteger/7-ReverseInteger.js
// Last updated: 7/6/2026, 11:44:18 PM
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let rev = 0;
    let sign = x < 0 ? -1 : 1;
    x = Math.abs(x);

    while (x > 0) {
        let digit = x % 10;
        x = Math.floor(x / 10);

        // Check for overflow before adding digit
        if (rev > Math.floor((2**31 - 1) / 10)) {
            return 0;
        }

        rev = rev * 10 + digit;
    }

    return rev * sign;
};